doc/source/whatsnew/v0.23.0.txt

```diff
@@ -462,6 +461,7 @@ Numeric
 - Bug in :class:`DatetimeIndex`, :class:`TimedeltaIndex` addition and subtraction of zero-dimensional integer arrays gave incorrect results (:issue:`19012`)
 - Bug in :func:`Series.__add__` adding Series with dtype ``timedelta64[ns]`` to a timezone-aware ``DatetimeIndex`` incorrectly dropped timezone information (:issue:`13905`)
 - Bug in :func:`Timedelta.__floordiv__` and :func:`Timedelta.__rfloordiv__` dividing by many incompatible numpy objects was incorrectly allowed (:issue:`18846`)
+- Bug in :func:`Timestamp.floor` where time stamps far in the future and past were not rounded correctly (:issue:`19206`)
```

Comment:
see below, need to add for ceil and for DatetimeIndex floor/ceil as well.

Reply:
need tests for DTI as well (these are in pandas/tests/indexes/datetimes/test_ops pls parametrize

Reply:
I suspect you need a separate fix there as well.